lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -101,22 +116,35 @@ def puppet4_repo(full_version)
       end
 
       def common_repos(full_version)
-        repos_arrary = if ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
-                         ["rhel-#{el_major_version}-server-satellite-maintenance-6-beta-rpms",
-                          "rhel-#{el_major_version}-server-satellite-tools-6-beta-rpms"]
-                       else
-                         ["rhel-#{el_major_version}-server-satellite-maintenance-6-rpms",
-                          "rhel-#{el_major_version}-server-satellite-tools-#{full_version}-rpms"]
-                       end
-
-        return repos_arrary.first(1) if feature(:satellite)
-
-        repos_arrary
+        sat_maint_version = if version(full_version) >= version('7.0')
+                              full_version
+                            else
+                              '6'
+                            end
+
+        # rubocop:disable Metrics/LineLength
+        repos = if el7? && use_beta_repos?
+                  ["rhel-#{el_major_version}-server-satellite-maintenance-#{sat_maint_version}-beta-rpms"]
+                elsif el7?
+                  ["rhel-#{el_major_version}-server-satellite-maintenance-#{sat_maint_version}-rpms"]
+                elsif use_beta_repos?
+                  ["satellite-maintenance-#{sat_maint_version}-beta-for-rhel-#{el_major_version}-x86_64-rpms"]
```

Comment:
will it be `satellite-maintenance-7-beta-for-rhel-8-x86_64-rpms` or `satellite-maintenance-7.0-beta-for-rhel-8-x86_64-rpms`? I would guess the former?

Reply:
I think the latter since the repository for maintenance is now versioned per Satellite version

Reply:
sure, but also for *satellite* itself, the beta repo was always "6-beta", whichever beta we shipped in there (I don't mind either way, just pointing out history)

Reply:
Ohh you are saying it was a one size fits all beta repository. That is indeed a question I do not know the answer to.

Reply:
I've updated it now to use `-7-beta-` to match the other names, but can adjust later, of course.